File: LeetCode/Easy/0938-range-sum-of-bst/0938-range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        if (root == null) return 0;

        int ret = 0;

        if (root.val >= low && root.val <= high) ret += root.val;

        if (root.val <= high) ret += rangeSumBST(root.right, low, high);
        if (root.val >= low) ret += rangeSumBST(root.left, low, high);
        
        return ret;
    }
}

File: LeetCode/Easy/0938-range-sum-of-bst/0938-range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        ret = [0]
        
        def dfs(node):
            if not node:
                return
            
            if low <= node.val <= high:
                ret[0] += node.val
            
            dfs(node.left)
            dfs(node.right)
        
        dfs(root)
        
        return ret[0]